bin/program.py

```diff
@@ -419,6 +419,10 @@ def add_callback(problem, path, c):
             problem._program_callbacks[path] = []
         problem._program_callbacks[path].append(c)
 
+    @staticmethod
```

Comment:
I think making this non-static is a bit nicer; `self.timeout()` reads better than `Program.timeout()`, and `timeout` feels more like a per program thing than a global thing anyway.

Actually, it's a bit weird that all programs have a timeout, but submissions actually use the problem timeout instead of the program timeout, but 'fixing' that (by propagating the timeout from problems to submissions) probably adds more code than it removes so not worth it I think.

Reply:
Given that `timeout` is, as I agree, "more like a per program thing than a global thing", I've inlined the function into `Generator.run` and `Visualizer.run`. In the parameter descriptions of the argument parser, the default value is duplicated as well, anyway. :stuck_out_tongue: (57be5a8)

Related to your other comment about rethinking Program timeouts vs. Submission timeouts, we can probably better open a new issue for that for now :stuck_out_tongue: